src/encoded/static/components/facets/defaults.js

```diff
@@ -416,6 +417,24 @@ export class DefaultDateSelectorFacet extends React.Component {
         const field = this.state.activeFacet;
         const activeFacet = results.facets.filter(f => f.field === this.state.activeFacet)[0];
 
+        // We have two facets registered for the date selector facet, one for date_released and one for date_submitted, but we only want to display one
+        // "displayField", when false, hides the duplicate facet when needed (most of the time)
+        // If we have date_released or date_submitted data only, then only one facet will be displayed and we don't need to hide one
+        // If we have both, we will just display the facet for date_released
+        if (facet.field === 'date_submitted' && results.facets.filter(f => f.field === 'date_released').length > 0) {
+            return null;
+        }
+        // If we are missing data for "date_submitted" or "date_released", we want to disable that radio button
+        // "missingField" indicates if there is a field to disable, and if so, which one
+        let missingField = null;
+        // Check if we are displaying date_submitted facet and there is no date_released data
+        if (facet.field === 'date_submitted' && !(results.facets.filter(f => f.field === 'date_released')).length > 0) {
```

Comment:
Technically, since you already tested for both date_submitted and date_released existing and then returned null in that case, then if you get to this line and `facet.field` is date_submitted, you’ll know that no date_released exists, guaranteed.

Reply:
I noticed this could be simplified at some point and then forgot to go back and take care of it. Thank you for reminding me!